packages/slate-editor/src/components/OptionsGroup/Option.tsx

```diff
@@ -0,0 +1,53 @@
+import classNames from 'classnames';
+import * as React from 'react';
+
+import styles from './OptionsGroup.module.scss';
+
+export interface OptionsGroupOption<T extends string> {
+    label: string;
+    value: T;
+    Icon?: React.ComponentType<React.SVGProps<SVGSVGElement> & { isActive: boolean }>;
+}
+
+interface Option<T> {
+    selected?: T;
+    onChange?: (value: T | undefined) => void;
+}
+
+export interface RadioChoose<T extends string> extends Option<T> {
+    type?: 'radio';
+}
+
+type ChooseOptionProps<T extends string> = RadioChoose<T> & {
+    name: string;
+    option: OptionsGroupOption<T>;
+    allSelected: Set<T | undefined>;
+};
+
+export function ChooseOption<T extends string>(props: ChooseOptionProps<T>) {
+    const onChange = React.useCallback(() => {
+        props.onChange?.(props.option.value);
+    }, [props.option, props.onChange]);
+
+    const isActive = props.allSelected.has(props.option.value);
```

Comment:
I see no reason to pass whole `allSelected` set from above if we only need to check if it's included.
A much more expected component API would be to only ask for `isSelected` prop from outside.
Would you agree?